### ISSUE TITLE
Focus on useMenu elements with mouse hover does not work

### ISSUE THREAD

**Issue by Slooowpoke** [adobe/react-spectrum#1301]
# 🐛 Bug Report 

Essentially, the example [here](https://react-spectrum.adobe.com/react-aria/useMenu.html) works fine. Hovering with the mouse of useMenu items works and applies focus. 

Inside Storybook, Codepen and our application (I think the similarity is they all use React-Router perhaps?) after page load when a mouse moves over the menu options, the focus is not applied. Keyboard inputs work fine and clicking with the mouse will apply focus, but not hover. However! If you navigate away and come back, it works fine? It's almost as if the initial page load mount is bad?

## 🤔 Expected Behavior

Page loads, component loads, user hovers with mouse over menu items, menu items have focus applied.

## 😯 Current Behavior

Page loads, component loads, user hovers with mouse over menu items, menu items don't have focus applied.

## 💁 Possible Solution

I wish I had a solution or a workaround.

## 🔦 Context

A basic profile menu with hovers, I can achieve the same with CSS but I figured if it works in the docs it should work in our application.

## 💻 Code Sample

https://codesandbox.io/s/react-aria-focus-bugs-fyto8

vs

https://react-spectrum.adobe.com/react-aria/useMenu.html

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-aria | 3.1.1 |
| react-stately   | 3.0.1
| Browser          | Chrome
| Operating System | Mac

## 🧢 Your Company/Team

N/A

**Comment by snowystinger** [comment]
Some potential issues I see might be happening.
1) A mismatch of versions, could you try using the `react-aria` and `react-stately` packages instead of the individual ones?
2) You're on react 17, could you try with a recent 16 version instead? our docs are built with 16.13.0
3) I added a console log to your render, it does seem to cause a re-render on mouse over, but I haven't dug far enough to see what is being updated

If you're trying to use the packages individually, it may be helpful to check in the network tab and see what versions codesandbox is resolving to.

I'll try to dig a little deeper when I have more time as well.

**Comment by Slooowpoke** [comment]
Heya, thanks for looking into that.

I kinda just cobbled together those packages as I was in a rush. Locally in our application we use React-Aria, React-Stately and React 16 and the issues occurs but I've also updated the [codepen](https://codesandbox.io/s/react-aria-focus-bugs-fyto8
) in the original issue with those packages.

It's odd because it does cause a re-render but everytime ```isFocused``` is false, I've also added a button to the codepen to check whether it was mounting / unmounting behaviour but I couldn't see anything odd.
